### ISSUE TITLE
Pixel 8 Issue - UnsatisfiedLinkError

### ISSUE THREAD

**Issue by mtwarner** [facebook/SoLoader#119]
We are getting the following on all Pixel 8 devices, all other devices seem to work just fine with our build. It does not appear to be an OS 14 issues as it loads fine on a Pixel 7 Pro with OS 14.

Exception java.lang.UnsatisfiedLinkError:
  at com.facebook.soloader.SoLoader.loadLibraryBySoName (SoLoader.java:455)

Our currently release version forces 0.10.3 but we have also tried the latest 0.10.5 to no avail, still crashing on load of our application.

Please advise.

Thanks - MTW

**Comment by andrewfluck** [comment]
We're experiencing the same thing on our end with one of our products too... I'm probably gonna list out some more info here too.

**Comment by shashachu** [comment]
I upgraded to a Pixel 8 and am seeing this same crash with an app we need for work. I'm not the developer but when it crashes repeatedly I can tap send feedback and I see this same callstack. In this case, it's crashing initializing libhermes.so for React Native.

**Comment by andrewfluck** [comment]
@shashachu @mtwarner We're (regrettably) stuck on an old version of react native, I was helping a dev on our team fix this by forcing version `0.9.0`. This seems to work. You can force dependency versions via the top level `build.gradle` file via the `ResolutionStrategy` api.

**Comment by mtwarner** [comment]
The issue we are seeing seems to be due to the facebook conceal encryption library, forcing version 0.9.0 does not work for us. I am at a loss right now except to get a new encryption library. What I don't understand is why is only this one device?

**Comment by ddinkov10** [comment]
If the problem is with the App Bundles i would suggest you to try this option
android.bundle.enableUncompressedNativeLibs = false in gradle.properties

[AGP 3.3.0](https://developer.android.com/build/releases/past-releases/agp-3-3-0-release-notes)

**Comment by andrewfluck** [comment]
> The issue we are seeing seems to be due to the facebook conceal encryption library, forcing version 0.9.0 does not work for us. I am at a loss right now except to get a new encryption library. What I don't understand is why is only this one device?

Hmmmm, I guess theres a multitude of issues that could cause this then... Using 0.9.0 worked for us, we're about to try 0.10.4 to make sure it works as well.

**Comment by ashwanihundwani1986** [comment]
@andrewfluck Did upgrading SoLoader worked for you?

**Comment by andrewfluck** [comment]
> @andrewfluck Did upgrading SoLoader worked for you?

Yes forcing the version worked for us. We're already using this version on one of our apps. A legacy app that was being maintained wasn't so I figured the logical step would be to update it. Turns out Google put out the same advice too.
